src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -51,18 +52,26 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
             IAsynchronousOperationListenerProvider listenerProvider)
             : base(threadingContext)
         {
-            AssertIsForeground();
-
             if (serviceProvider != null)
             {
-                var textManager = (IVsTextManager2)serviceProvider.GetService(typeof(SVsTextManager));
-                if (textManager.GetExpansionManager(out _expansionManager) == VSConstants.S_OK)
-                {
-                    ComEventSink.Advise<IVsExpansionEvents>(_expansionManager, this);
-                    _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
-                    _languageGuidForSnippets = languageGuidForSnippets;
-                    PopulateSnippetCaches();
```

Comment:
📝 I was more concerned about this change before I eventually realized that the _implementation_ of `PopulateSnippetCaches` was already operating in a fire-and-forget asynchronous manner. The new code doesn't change this fact, but rather brings it to the surface.